tests/models/gemma/test_modeling_gemma.py

```diff
@@ -615,6 +615,44 @@ def test_model_2b_flash_attn(self):
 
         self.assertEqual(output_text, EXPECTED_TEXTS)
 
+    @require_read_token
+    def test_model_2b_pipeline_flex_attention(self):
+        # See https://github.com/huggingface/transformers/pull/31747 -- pipeline was broken for Gemma2 before this PR
+        model_id = "google/gemma-2b"
+        # EXPECTED_TEXTS should match the same non-pipeline test, minus the special tokens
+        EXPECTED_TEXTS = [
+            "Hello I am doing a little research on the subject of the 2019-01-09",
+            "Hi today I am going to be honest, I’m not sure if you’ve ever heard of the",
+        ]
+        model = AutoModelForCausalLM.from_pretrained(
+            model_id, low_cpu_mem_usage=True, attn_implementation="flex_attention"
+        ).to(torch_device)
```

Comment:
Just a small addition I like is to check if the model still has flex attn set, i.e. assert `model.config._attn_implementation == "flex_attn"`